v-next/hardhat/src/internal/builtin-plugins/network-manager/json-rpc-request-modifiers/accounts/automatic-sender-provider.ts

```diff
@@ -8,20 +8,23 @@ import { Sender } from "./sender.js";
  * ensuring dynamic selection of the sender for all JSON-RPC requests without requiring manual input.
  */
 export class AutomaticSender extends Sender {
+  #alreadyFetchedAccounts = false;
   #firstAccount: string | undefined;
 
-  protected async getSender(): Promise<string> {
-    if (this.#firstAccount === undefined) {
+  protected async getSender(): Promise<string | undefined> {
+    if (this.#alreadyFetchedAccounts === false) {
       const accounts = await this.provider.request({
         method: "eth_accounts",
       });
 
+      // TODO: This shouldn't be an exception but a failed JSON response!
```

Comment:
This drove me crazy tbh. Throwing here makes things much more complex, as viem gets confused.

Reply:
Is this an issue on Viem's side?

Reply:
I'm confused here: is this TODO for later or it should be implemented in this PR? If it's for later, is there an issue tracking it?

Reply:
@ChristopherDedominici: later

@jxom, I don't think this is an issue. It's just that we used to have a bug where this code entered an infinite recursion, and viem was doing many retries, so I spent a long time thinking that the other bug appeared again 😅

Is there a way to throw from a provider so that viem just bubbles/ignores the error?

Reply:
You can pass `retryCount: 0` as second param